File: 1512-number-of-good-pairs/1512-number-of-good-pairs.py
class Solution:
    def numIdenticalPairs(self, a: List[int]) -> int:
        n = len(a)
        ans = 0
        for i in range(n):
            for j in range(i+1, n):
                if a[i] == a[j]:
                    ans += 1
        return ans

File: 1512-number-of-good-pairs/1512-number-of-good-pairs.cpp
class Solution {
public:
    int numIdenticalPairs(vector<int>& a) {
        vector<int> c(101);
        int ans = 0;
        for (auto x : a) {
            ans += c[x];
            c[x]++;
        }
        return ans;
    }
};